ravenframework/Optimizers/GeneticAlgorithm.py

```diff
@@ -322,20 +612,45 @@ def handleInput(self, paramInput):
       @ Out, None
     """
     RavenSampled.handleInput(self, paramInput)
-    # GAparams
+    ####################################################################################
+    # GAparams                                                                         #
+    ####################################################################################
     gaParamsNode = paramInput.findFirst('GAparams')
-    # populationSize
+
+    ####################################################################################
+    # populationSize                                                                   #
+    ####################################################################################
     populationSizeNode = gaParamsNode.findFirst('populationSize')
     self._populationSize = populationSizeNode.value
-    # parent selection
+
+    ####################################################################################
+    # parent selection node                                                            #
+    ####################################################################################
     parentSelectionNode = gaParamsNode.findFirst('parentSelection')
     self._parentSelectionType = parentSelectionNode.value
     self._parentSelectionInstance = parentSelectionReturnInstance(self, name=parentSelectionNode.value)
-    # reproduction node
+
+    if self._isMultiObjective and self._parentSelectionType != 'tournamentSelection':
+      self.raiseAnError(IOError, f'Currently, "tournamentSelection" in the only <parentSelection> mechanism supported by the multi-objective Genetic Algorithms.')
+
+    ####################################################################################
+    # reproduction node                                                                #
+    ####################################################################################
     reproductionNode = gaParamsNode.findFirst('reproduction')
     self._nParents = int(np.ceil(1/2 + np.sqrt(1+4*self._populationSize)/2))
     self._nChildren = int(2*comb(self._nParents,2))
-    # crossover node
+
+    ####################################################################################
+    # k-Selection node                                                                #
+    ####################################################################################
+    if reproductionNode.findFirst('kSelection') is None:
+      self._kSelection = 3 # Default value is set to 3.
```

Comment:
Please define it in the __init__ method, and provide a default value in the inputSpecs. If you do so, I think you do not need the if else block here.

Reply:
Fixed 280e525a53be20c91a6e4a281cdecb27d884f80f